power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -81,6 +81,14 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     using GetIndexerFunc = void (*)(MainModelImpl const& x, ID const* id_begin, Idx size, Idx* indexer_begin);
 
    public:
+    struct cached_update_t {
+        static constexpr auto cached_update = true;
+    };
```

Comment:
Nice to use a tag type so it is more clear. As a convention for tag type with boolean value, we just inherit from `std::true_type` or `std::false_type`.

```suggestion
    struct cached_update_t : std::true_type {};
```